docs/guide.rst

```diff
@@ -54,6 +54,37 @@ be between 0.0 and 1.0 inclusive. Vertices out of this range in NDC
 will not introduce any errors, but they will be clipped.
 
 
+Communicating array data
+------------------------
+
+The wgpu-py library makes no assumptions about how you store your data.
+In places where you provide data to the API, it can consume any data
+that supports the buffer protocol, which includes ``bytes``,
+``bytearray``, ctypes arrays, and numpy arrays.
+
+In places where data is returned, the API returns a ``memoryview``
+object. These objects provide a quite versatile view on ndarray data:
+
+.. code-block:: py
+
+    # One could, for instance read the content of a buffer
+    m = buffer.map(wgpu.MapMode.READ)
+    buffer.unmap()
+    # Cast it to float32
+    m = m.cast("f")
+    # Index it
+    m[0]
+    # Show the content
+    print(m.tolist())
+
+Chances are that you prefer Numpy. Converting the ``memoryview`` to a
+numpy array (without copying the data) is easy:
+
+.. code-block:: py
+
+    array = np.frombuffer(m, np.float32)
```

Comment:
This summarizes this PR pretty well.

Reply:
And it feels really good.

Reply:
I don't filly understand why you need to map and then unmap but the rest of it looks great! Very simple.

Reply:
ehr ... I double-checked, and as soon as you `unmap` gets called, the memory is no longer valid. An oversight on my part. We need to tweak this API a bit, because this is way it is much to easy for users to shoot themselves in the foot ...